enterprise/frontend/src/metabase-enterprise/content_verification/metrics.tsx

```diff
@@ -0,0 +1,83 @@
+import { type ChangeEvent, useCallback } from "react";
+import { t } from "ttag";
+
+import type {
+  MetricFilterControlsProps,
+  MetricFilterSettings,
+} from "metabase/browse/utils";
+import { useUserSetting } from "metabase/common/hooks";
+import { getSetting } from "metabase/selectors/settings";
+import { Button, Icon, Paper, Popover, Switch, Text } from "metabase/ui";
+import type { State } from "metabase-types/store";
+
+const USER_SETTING_KEY = "browse-filter-only-verified-metrics";
+
+export function getDefaultMetricFilters(state: State): MetricFilterSettings {
+  return {
+    verified: getSetting(state, USER_SETTING_KEY) ?? false,
+  };
+}
+
+export const MetricFilterControls = ({
+  metricFilters,
+  setMetricFilters,
+}: MetricFilterControlsProps) => {
+  const areAnyFiltersActive = Object.values(metricFilters).some(Boolean);
+
+  const [_, setUserSetting] = useUserSetting(USER_SETTING_KEY);
+
+  const handleVerifiedFilterChange = useCallback(
+    function (evt: ChangeEvent<HTMLInputElement>) {
+      setMetricFilters({ ...metricFilters, verified: evt.target.checked });
```

Comment:
In `enterprise/frontend/src/metabase-enterprise/content_verification/ModelFilterControls.tsx`, the function `setVerifiedFilterStatus`, which I think is analogous to `setMetricFilters`, is debounced. This avoids a problem where clicking the filter a few times in a row makes the page behave oddly. Might it be worth debouncing `setMetricFilters` for the same reason?

Reply:
I'm not seeing any odd behavior when clicking the filter a few times (very fast even)! Could it be that the issue is not present for Metrics?

Reply:
yeah the metric implementation is slightly different and doesn't have this issue

Reply:
Excellent!